packages/discovery/src/cli/layoutCommand.ts

```diff
@@ -0,0 +1,45 @@
+import { Logger } from '@l2beat/backend-tools'
+import { writeFileSync } from 'fs'
+
+import { DiscoveryCliConfig } from '../config/config.discovery'
+import { flattenLayout } from '../layout/flattenLayout'
+import { mergeFlatLayouts } from '../layout/mergeFlatLayouts'
+import { parseAndGetLayout } from '../layout/parseAndGetLayout'
+import { EthereumAddress } from '../utils/EthereumAddress'
+import { EtherscanLikeClient } from '../utils/EtherscanLikeClient'
+import { HttpClient } from '../utils/HttpClient'
+
+export async function layoutCommand(
+  config: DiscoveryCliConfig,
+  logger: Logger,
+): Promise<void> {
+  if (!config.layout) {
+    return
+  }
+  const http = new HttpClient()
+  const etherscanClient = EtherscanLikeClient.createForDiscovery(
+    http,
+    config.chain.etherscanUrl,
+    config.chain.etherscanApiKey,
+    config.chain.etherscanUnsupported,
+  )
+  await runLayout(etherscanClient, config.layout.addresses, logger)
+}
+
+async function runLayout(
```

Comment:
What do you think about making this function return the results (the `layout` variable) and the function above can save them. This would allow us to later export this `runLayout` and use the results without having to interface with the filesystem